warehouse-management-service/pkg/database/postgres/shelf_block.go

```diff
@@ -0,0 +1,200 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	wms "warehouse-management-service"
+)
+
+type shelfBlockQueries interface {
+	createShelfBlockTx(ctx context.Context, tx *sql.Tx, block wms.ShelfBlock) error
+	getShelfBlockByIdTx(ctx context.Context, tx *sql.Tx, id string) (wms.ShelfBlock, error)
+	updateShelfBlockTx(ctx context.Context, tx *sql.Tx, block wms.ShelfBlock) error
+	deleteShelfBlockTx(ctx context.Context, tx *sql.Tx, id string) error
+	warehouseExistsTx(ctx context.Context, tx *sql.Tx, id string) (bool, error)
+}
+
+type shelfBlockQueriesImpl struct{}
+
+type ShelfBlockService struct {
+	queries shelfBlockQueries
+	db      *sql.DB
+}
+
+var InvalidWarehouse = errors.New("invalid warehouseId")
```

Comment:
Let's not make a static variable for an error. Ideally, there should be a new call to `errors.New` whenever we want to return an error. 

From the [docs](https://pkg.go.dev/errors#New):
> Each call to New returns a distinct error value even if the text is identical

Reply:
An `error` can have other meta information which will be lost if we just use a static definition.

Reply:
This error is returned when the validation of a foreign key relation fails, it does not replace any underlying error (check [usage](https://github.com/nilenso/siris-onboarding/pull/31/files#diff-d40dcbfb80511563d7f2e94f43c6376b948ffbb424e9edb560244c085a652f2fR112))

Moreover, having package-level error vars is a common pattern to define custom errors, as seen in the [stdlib](https://github.com/golang/go/blob/master/src/io/io.go) too. 

However, in a later refactor, I am planning on removing this error abstraction and instead parsing the SQL library error itself.

Reply:
>Moreover, having package-level error vars is a common pattern to define custom errors, as seen in the [stdlib](https://github.com/golang/go/blob/master/src/io/io.go) too.

Nice. TIL!